app/shared/github_handler.rb

```diff
@@ -42,6 +44,24 @@ def self.included(klass)
 
     def github_action(client, &block)
       if client.kind_of?(Octokit::Client)
+        # Inject Faraday::Request::Retry to the client if necessary
+        unless client.middleware.handlers.include?(Faraday::Request::Retry)
+          middleware_dup = client.middleware.dup
+          client.middleware = middleware_dup
+          client.middleware.insert_before(Faraday::Adapter::NetHttp,
```

Comment:
This is a style nitpick, but could you please put the first parameter on a new line, and indent the parameters 2 spaces out?

i.e.,

```ruby
client.middleware.insert_before(
  Faraday::Adapter::NetHttp,
  Faraday::Request::Retry,
  # ...
)
```